src/types/events/index.ts

```diff
@@ -24,6 +25,17 @@ export interface SlackEventMiddlewareArgs<EventType extends string = string> {
   say: WhenEventHasChannelContext<this['payload'], SayFn>;
 }
 
+export interface SlackSubtypedEventMiddlewareArgs<
+  EventType extends string = string,
+  EventSubtype extends string | undefined = undefined
+> {
+  payload: EventSubtype extends '*' ? EventFromType<EventType> : EventFromTypeAndSubtype<EventType, EventSubtype>;
```

Comment:
To keep backward compatibility, we are unable to use undefined for subtype here.